homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+            abort(500)
+
+        json_response = [{'success': {'username': '12345678901234567890'}}]
+
+        return Response(json.dumps(json_response), mimetype='application/json')
+
+
+def parse_hue_api_put_light_body(request_json, entity):
+    """Parse the body of a request to change the state of a light."""
+    if 'on' in request_json:
+        if not isinstance(request_json['on'], bool):
+            return None
+
+        if request_json['on']:
+            # Echo requested device be turned on
+            brightness = 100
+            report_brightness = False
+            result = True
+        else:
+            # Echo requested device be turned off
+            brightness = 0
+            report_brightness = False
+            result = False
+
+    if 'bri' in request_json:
+        # Make sure the entity actually supports brightness
+        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        if (entity_features & SUPPORT_BRIGHTNESS) == SUPPORT_BRIGHTNESS:
+            try:
+                # Clamp brightness from 0 to 255
+                brightness = max(0, min(int(request_json['bri']), 255))
+            except ValueError:
+                return None
+
+            report_brightness = True
+            result = True if brightness > 0 else False
+
+    return (result, brightness) if report_brightness else (result, None)
+
+
+def entity_to_json(entity, is_on=None, brightness=None):
+    """Convert an entity to its JSON representation."""
+    if is_on is None:
+        is_on = entity.state == "on"
+
+    if brightness is None:
+        brightness = 255 if is_on else 0
+
+    name = entity.attributes.get(
+        ATTR_ECHO_FRIENDLY_NAME, entity.attributes['friendly_name'])
+
+    return {
+        'state':
+        {
+            'on': is_on,
+            'bri': brightness,
+            'hue': 0,
+            'sat': 0,
+            'effect': 'none',
+            'ct': 0,
+            'alert': 'none',
+            'reachable': True
+        },
+        'type': 'Dimmable light',
+        'name': name,
+        'modelid': 'LWB004',
+        'manufacturername': 'Philips',
+        'uniqueid': entity.entity_id,
+        'swversion': '66012040'}
+
+
+def create_hue_success_response(entity_id, attr, value):
+    """Create a success response for an attribute set on a light."""
+    success_key = '/lights/{}/state/{}'.format(entity_id, attr)
+    return {'success': {success_key: value}}
+
+
+class UPNPResponderThread(threading.Thread):
+    """Handle responding to UPNP/SSDP discovery requests."""
+
+    stop_thread = False
+
+    def __init__(self, host_ip_addr, listen_port):
+        """Initialize the class."""
+        threading.Thread.__init__(self)
+
+        self.host_ip_addr = host_ip_addr
+        self.listen_port = listen_port
+
+        # Note that the double newline at the end of
+        # this string is required per the SSDP spec
+        resp_template = """HTTP/1.1 200 OK

```

Comment:
You should be able to do this with the built in Flask/`http` tools. I believe this is just a list of a bunch of headers and the 200 code.